blockchain/chain/src/store/chain_store.rs

```diff
@@ -731,14 +731,13 @@ where
 }
 
 /// Returns the genesis block from storage.
-pub fn genesis<DB>(db: &DB) -> Result<Option<BlockHeader>, Error>
+pub fn genesis<DB>(db: &DB) -> Result<BlockHeader, Error>
 where
     DB: Blockstore + Store,
 {
-    Ok(db
-        .read(GENESIS_KEY)?
-        .map(|bz| BlockHeader::unmarshal_cbor(&bz))
-        .transpose()?)
+    db.read(GENESIS_KEY)?
+        .map(|bz| BlockHeader::unmarshal_cbor(&bz).map_err(|e| Error::Other(e.to_string())))
+        .ok_or(Error::Other("no genesis".to_string()))?
```

Comment:
Maybe the error message could be `Genesis key not defined in database`

Reply:
Changed